diff --git a/meson.build b/meson.build
index 039e7e13..ef86a3cf 100644
--- a/meson.build
+++ b/meson.build
@@ -173,7 +173,8 @@ group_internal = get_option('programmer').contains('group_internal')
 group_external = get_option('programmer').contains('group_external')
 
 libpci     = dependency('libpci', required : group_pci, version : '>=2.2.0',
-  static : (host_machine.system() == 'openbsd' ? true : false)) # On openbsd a static version of libpci is needed to get also -libz
+  static : ((host_machine.system() == 'openbsd') or get_option('prefer_static') ? true : false))
+# On openbsd a static version of libpci is needed to get also -libz
 libusb1    = dependency('libusb-1.0', required : group_usb)
 libftdi1   = dependency('libftdi1', required : group_ftdi)
 libjaylink = dependency('libjaylink', required : group_jlink, version : '>=0.3.0')
